Linked list with void pointers.c

```diff
@@ -37,37 +34,97 @@ void display_float(node *tail)
 {
     printf("%lf\n", *(double *)tail->data);
 }
-void display(void(*display_datatype)(node *tail), node *tail)
+void display_char(node *tail)

```

Comment:
Why should display methods know about full node info ? It uses only data ptr right ?

convert it into display_char(void *ptr);